src/Web/Rsp.IrasPortal.Web/Controllers/QuestionSetController.cs

```diff
@@ -1,287 +1,215 @@
-        using var stream = file.OpenReadStream();
-        using var reader = ExcelReaderFactory.CreateReader(stream);
-        var result = reader.AsDataSet(new ExcelDataSetConfiguration()
-        {
-            ConfigureDataTable = (_) => new ExcelDataTableConfiguration()
-            {
-                UseHeaderRow = true
-            }
-        });
+        var fileProcessResponse = questionSetService.ProcessQuestionSetFile(file);
 
-        List<DataTable> moduleSheets = [];
-        DataTable rulesSheet = new();
-        DataTable answerOptionsSheet = new();
-        DataTable contentsSheet = new();
-
-        foreach (var sheetName in SheetNames.All)
+        if (fileProcessResponse.Error != null)
         {
-            var sheet = result.Tables[sheetName];
-
-            if (sheet == null)
-            {
-                ModelState.AddModelError("Upload", $"Sheet '{sheetName}' cannot be found");
-                continue;
-            }
-
-            if (HasValidColumns(sheetName, sheet))
-            {
-                if (sheetName == SheetNames.Rules)
-                {
-                    rulesSheet = sheet;
-                }
-                else if (sheetName == SheetNames.AnswerOptions)
-                {
-                    answerOptionsSheet = sheet;
-                }
-                else if (sheetName == SheetNames.Contents)
-                {
-                    contentsSheet = sheet;
-                }
-                else
-                {
-                    moduleSheets.Add(sheet);
-                }
-            }
+            ModelState.AddModelError("Upload", fileProcessResponse.ReasonPhrase!);
+            return View(nameof(Index), model);
         }
 
-        if (ModelState.ErrorCount > 0)
+        if (!fileProcessResponse.IsSuccessStatusCode)
         {
-            return View(model);
+            ModelState.AddModelError("Upload", "An unknown error occured");
+            return View(nameof(Index), model);
         }
 
-        foreach (DataRow category in contentsSheet.Rows)
-        {
-            var categoryId = category.Field<string>(ContentsColumns.Tab);
+        model.QuestionSetDto = fileProcessResponse.Content;
 
-            if (categoryId == null || !SheetNames.Modules.Contains(categoryId))
-            {
-                continue;
-            }
+        if (!await ValidateQuestions(model))
+        {
+            return View(nameof(Index), model);
+        }
 
-            var categoryDto = new CategoryDto
-            {
-                CategoryId = categoryId,
-                CategoryName = category.Field<string>(ContentsColumns.Category) ?? "",
-            };
+        var fileUploadResponse = await questionSetService.AddQuestionSet(fileProcessResponse.Content!);
 
-            model.CategoryDtos.Add(categoryDto);
+        if (!fileUploadResponse.IsSuccessStatusCode)
+        {
+            ModelState.AddModelError("Upload", "Internal server error");
```

Comment:
use nameof(Upload) instead of "Upload"
```suggestion
            ModelState.AddModelError(nameof(Upload), "Internal server error");
```